packages/core-mobile/app/store/account/thunks.ts

```diff
@@ -24,18 +25,25 @@ export const addAccount = createAsyncThunk<void, void, ThunkApi>(
     const isDeveloperMode = selectIsDeveloperMode(state)
     const activeNetwork = selectActiveNetwork(state)
     const activeWalletId = selectActiveWalletId(state)
+
     if (!activeWalletId) {
       throw new Error('Active wallet ID is not set')
     }
     const wallet = selectWalletById(activeWalletId)(state)
     if (!wallet) {
       throw new Error('Wallet not found')
     }
+
+    const wallets = selectWallets(state)
+    const allAccountsCount = Object.values(wallets).reduce((acc, item) => {
+      const accounts = selectAccountsByWalletId(item.id)(state)
+      return acc + accounts.length
+    }, 0)
```

Comment:
I think you can also just select all accounts in the store here instead of grabbing the wallets -> accounts